File: jobs/2026-05-27__05-16-59/llamacloud_classify_documents_ts__7YK4v5j/artifacts/user/myproject/classify.ts
import LlamaCloud from '@llamaindex/llama-cloud';
import * as fs from 'fs';
import * as path from 'path';

async function main() {
    const client = new LlamaCloud();
    
    const sampleDir = path.join(process.cwd(), 'samples');
    const files = ['invoice.txt', 'receipt.txt', 'contract.txt'];
    
    // Upload files
    const fileIds: string[] = [];
    const fileIdToName = new Map<string, string>();
    
    for (const filename of files) {
        const filePath = path.join(sampleDir, filename);
        const fileStream = fs.createReadStream(filePath);
        const uploadedFile = await client.files.create({
            file: fileStream,
            purpose: 'classify'
        });
        fileIds.push(uploadedFile.id);
        fileIdToName.set(uploadedFile.id, filename);
    }
    
    // Submit classify job
    const result = await client.classifier.classify({
        file_ids: fileIds,
        rules: [
            { type: 'invoice', description: 'A commercial invoice with invoice number, bill-to section, line items, and totals' },
            { type: 'receipt', description: 'A point-of-sale receipt' },
            { type: 'contract', description: 'A legal services agreement with signatures' }
        ],
        mode: 'FAST'
    });
    
    // Write to output.log
    const runId = process.env.ZEALT_RUN_ID || 'unknown';
    const logLines: string[] = [];
    logLines.push(`Run ID: ${runId}`);
    
    // Create a map of file_id to result item for easy lookup
    const itemMap = new Map<string, any>();
    for (const item of result.items) {
        if (item.file_id) {
            itemMap.set(item.file_id, item);
        }
    }
    
    // Output in the same order as submitted
    for (const fileId of fileIds) {
        const item = itemMap.get(fileId);
        if (!item || !item.result) continue;
        
        const filename = fileIdToName.get(fileId) || 'unknown';
        const type = item.result.type;
        const confidence = item.result.confidence;
        logLines.push(`Classified: ${filename} | Type: ${type} | Confidence: ${confidence}`);
    }
    
    fs.writeFileSync('output.log', logLines.join('\n') + '\n');
}

main().catch(console.error);

File: jobs/2026-07-02__17-09-42/llamacloud_classify_documents_py__ZBb7dGM/artifacts/user/myproject/classify.py
#!/usr/bin/env python3
"""Classify a single PDF document using the LlamaCloud Classifier.

Usage:
    python3 classify.py <path-to-pdf>

Reads the API key from the LLAMA_CLOUD_API_KEY environment variable and prints
a single JSON line to stdout containing the classified ``type``, the
``confidence`` score reported by the classifier, and the resolved absolute
``file`` path. Exits with status 0 on success and a non-zero status on any
error (failed upload, classifier error, empty result, etc.).
"""

import argparse
import json
import os
import sys
from pathlib import Path

from llama_cloud import LlamaCloud


# Classification rules - order matters and must be preserved exactly.
RULES = [
    {
        "type": "invoice",
        "description": (
            "documents that contain an invoice number, invoice date, bill-to "
            "section, and line items with totals"
        ),
    },
    {
        "type": "receipt",
        "description": (
            "short purchase receipts, typically from POS systems, with "
            "merchant, items and total, often a single page"
        ),
    },
    {
        "type": "contract",
        "description": (
            "multi-section legal agreement with parties, terms, and signature "
            "lines"
        ),
    },
]


def parse_args(argv=None):
    parser = argparse.ArgumentParser(
        description="Classify a PDF document using the LlamaCloud Classifier.",
    )
    parser.add_argument(
        "file",
        help="Path to a local PDF document to classify.",
    )
    return parser.parse_args(argv)


def main(argv=None) -> int:
    args = parse_args(argv)

    file_path = Path(args.file).resolve()
    if not file_path.is_file():
        print(
            f"Error: file not found: {args.file}",
            file=sys.stderr,
        )
        return 2

    api_key = os.environ.get("LLAMA_CLOUD_API_KEY")
    if not api_key:
        print(
            "Error: LLAMA_CLOUD_API_KEY environment variable is not set.",
            file=sys.stderr,
        )
        return 3

    client = LlamaCloud(api_key=api_key)

    # Upload the file for classification.
    try:
        with open(file_path, "rb") as fh:
            uploaded = client.files.create(file=fh, purpose="classify")
    except Exception as exc:  # pragma: no cover - network/SDK errors
        print(f"Error: failed to upload file: {exc}", file=sys.stderr)
        return 4

    file_id = getattr(uploaded, "id", None)
    if not file_id:
        print("Error: upload succeeded but no file id was returned.",
              file=sys.stderr)
        return 5

    # Run the classifier in FAST mode with the three rules.
    try:
        response = client.classifier.classify(
            file_ids=[file_id],
            rules=RULES,
            mode="FAST",
        )
    except Exception as exc:  # pragma: no cover - network/SDK errors
        print(f"Error: classifier request failed: {exc}", file=sys.stderr)
        return 6

    items = getattr(response, "items", None) or []
    if not items:
        print("Error: classifier returned no items.", file=sys.stderr)
        return 7

    item = items[0]
    result = getattr(item, "result", None)
    if result is None:
        print("Error: classifier item has no result.", file=sys.stderr)
        return 8

    classified_type = getattr(result, "type", None)
    confidence = getattr(result, "confidence", None)

    if classified_type is None:
        print("Error: classifier result is missing a type.",
              file=sys.stderr)
        return 9

    output = {
        "type": classified_type,
        "confidence": confidence,
        "file": str(file_path),
    }
    print(json.dumps(output))
    return 0


if __name__ == "__main__":
    sys.exit(main())